### PR TITLE
Implement SEP-986: Tool name validation

### PR DESCRIPTION
**Pull request by felixweinberger**
## Motivation and Context

https://github.com/modelcontextprotocol/modelcontextprotocol/issues/986

Implements tool name validation according to SEP-986 specification. Tool names are validated at registration time in both FastMCP and low-level server paths, issuing warnings for non-conforming names without blocking registration.

Taking similar approach to reference implementation in TS: https://github.com/modelcontextprotocol/typescript-sdk/pull/900

## How Has This Been Tested?

Unit tests - 29 new tests for the validation module covering:
- Valid names (alphanumeric, underscores, dashes, dots, max length)
- Invalid names (empty, too long, invalid characters, unicode)
- Warning patterns (leading/trailing dash/dot)
- Edge cases

All existing tests continue to pass.

## Breaking Changes

None, but console warnings may appear for SDK users who have tool names that do not satisfy the SEP.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update

## Checklist

- [x] I have read the [MCP Documentation](https://modelcontextprotocol.io)
- [x] My code follows the repository's style guidelines
- [x] New and existing tests pass locally
- [x] I have added appropriate error handling
- [x] I have added or updated documentation as needed

### CODE REVIEW

File: src/mcp/shared/tool_name_validation.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
+"""Tool name validation utilities according to SEP-986.
+
+Tool names SHOULD be between 1 and 128 characters in length (inclusive).
+Tool names are case-sensitive.
+Allowed characters: uppercase and lowercase ASCII letters (A-Z, a-z),
+digits (0-9), underscore (_), dash (-), and dot (.).
+Tool names SHOULD NOT contain spaces, commas, or other special characters.
+
+See: https://github.com/modelcontextprotocol/modelcontextprotocol/issues/986
```
Comment (pcarleton) [inline src/mcp/shared/tool_name_validation.py]:
```suggestion
See: https://modelcontextprotocol.io/specification/draft/server/tools#tool-names
```

File: src/mcp/shared/tool_name_validation.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+import re
+from dataclasses import dataclass, field
+
+logger = logging.getLogger(__name__)
+
+# Regular expression for valid tool names according to SEP-986 specification
+TOOL_NAME_REGEX = re.compile(r"^[A-Za-z0-9._-]{1,128}$")
+
+# SEP reference URL for warning messages
+SEP_986_URL = "https://github.com/modelcontextprotocol/modelcontextprotocol/issues/986"
```
Comment (pcarleton) [inline src/mcp/shared/tool_name_validation.py:24]:
```suggestion
SEP_986_URL = "https://modelcontextprotocol.io/specification/2025-11-25/server/tools#tool-names"
```

File: src/mcp/shared/tool_name_validation.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
+"""Tool name validation utilities according to SEP-986.
+
+Tool names SHOULD be between 1 and 128 characters in length (inclusive).
+Tool names are case-sensitive.
+Allowed characters: uppercase and lowercase ASCII letters (A-Z, a-z),
+digits (0-9), underscore (_), dash (-), and dot (.).
+Tool names SHOULD NOT contain spaces, commas, or other special characters.
+
+See: https://modelcontextprotocol.io/specification/draft/server/tools#tool-names
```
Comment (pcarleton) [inline src/mcp/shared/tool_name_validation.py:9]:
```suggestion
See: https://modelcontextprotocol.io/specification/2025-11-25/server/tools#tool-names
```
